### PR TITLE
Add nginx image

### PR DESCRIPTION
**Pull request by esenti**
Simple image containing ngnix for serving static content.

### CODE REVIEW

**Comment by rollcat** [comment]
Why not https://github.com/unit9/docklabs/tree/master/kubenginx ?

**Comment by esenti** [comment]
I think it's a good idea to have a simple nginx image where you can just `COPY` your static content and have it served, without need for further configuration and with no k8s-specific parts. Though `kubenginx` would probably work too.

**Comment by rollcat** [comment]
You can do all of that with `kubenginx` too. It has some hooks to play nice with k8s, but I don't think there's anything that would stop you from using it anywhere else.

If we'd like a separate / simpler image, I'd say we should have a base nginx + customisations for specific things (like k8s).

**Comment by esenti** [comment]
We already have another nginx image, with PHP support: https://github.com/unit9/docklabs/tree/master/nginx-php5. But yes, it might be a good idea to create a common image and customize it for k8s/PHP.

**Comment by esenti** [comment]
Made both `nginx-php5` and `kubenginx` use use the base `nginx` image and add custom config to it.

**Review by rollcat — APPROVED** [review]
Cool! Can you please adjust the image build dependencies in all automated builds on Dockerhub before you merge?

File: kubenginx/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,6 @@
-FROM unit9/base:latest
+FROM unit9/nginx:latest
 MAINTAINER Kamil Cholewiński <kamil.cholewinski@unit9.com>
 
-RUN apt-get update && \
-    apt-get install --yes --no-install-recommends \
-            nginx-light \
-    && rm -rf /var/cache/apt /var/lib/apt/lists
 ADD nginx.run  /etc/service/nginx/run
 ADD rc.local   /etc/rc.local
```
Comment (rollcat) [inline kubenginx/Dockerfile]: Overrides nginx.rc from base nginx?

File: kubenginx/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,6 @@
-FROM unit9/base:latest
+FROM unit9/nginx:latest
 MAINTAINER Kamil Cholewiński <kamil.cholewinski@unit9.com>
 
-RUN apt-get update && \
-    apt-get install --yes --no-install-recommends \
-            nginx-light \
-    && rm -rf /var/cache/apt /var/lib/apt/lists
 ADD nginx.run  /etc/service/nginx/run
```
Comment (rollcat) [inline kubenginx/Dockerfile]: Overrides nginx.rc from base nginx?

File: nginx-php5/Dockerfile [hunk 3]
```diff
@@ -1,12 +1,11 @@
...
 MAINTAINER Artiom Vasiliev <artiom@unit9.com>
 
 COPY ["php-fpm.rc", "/etc/service/php-fpm/run"]
-COPY ["nginx.rc", "/etc/service/nginx/run"]
 
 RUN apt-get update && \
-    apt-get install -y nginx php5 php5-common php5-curl php5-fpm php5-gd \
+    apt-get install -y php5 php5-common php5-curl php5-fpm php5-gd \
         php5-mysql php5-xmlrpc php5-xcache php5-memcache && \
     rm -rf /var/cache/apt /var/lib/apt/lists
```
Comment (rollcat) [inline nginx-php5/Dockerfile:9]: Line 14 (sorry github doesn't allow me to post a comment there), it looks like the nginx.conf is installed as the default site, and nginx.conf from base nginx doesn't pull it

**Review by rollcat — CHANGES_REQUESTED** [review]
Sorry, thought it was ok, found some issues

File: kubenginx/Dockerfile (hunk 2 above)
Comment (esenti) [reply kubenginx/Dockerfile]: Right, no need to do this.

File: nginx-php5/Dockerfile (hunk 3 above)
Comment (esenti) [reply nginx-php5/Dockerfile:9]: The default nginx config (that the base uses) will pull it.

Comment (rollcat) [reply nginx-php5/Dockerfile:9]: Wait, I see the problem.

Everything is named nginx.conf, but ends up in various places that are not /etc/nginx/nginx.conf. Not good.

Some things are named nginx.run, or nginx.rc, or else, but they all end up in /etc/service/nginx/run. Not good.

Some things override other random things.

Let me fix ;)

Comment (rollcat) [reply nginx-php5/Dockerfile:9]: https://github.com/unit9/docklabs/commits/feature/nginx-proposed ?

Comment (esenti) [reply nginx-php5/Dockerfile:9]: Looks good, pulled the commits into this PR.

**Review by rollcat — APPROVED** [review]

**Pull request merged**